diesel/src/connection/mod.rs

```diff
@@ -95,10 +99,19 @@ pub trait Connection: SimpleConnection + Sized + Send {
         let transaction_manager = self.transaction_manager();
         transaction_manager.begin_transaction(self)?;
         match f() {
-            Ok(value) => {
-                transaction_manager.commit_transaction(self)?;
-                Ok(value)
-            }
+            Ok(value) => match transaction_manager.commit_transaction(self) {
```

Comment:
Is there any specific reason to duplicate the error handling code here and in the postgres specific transaction implementation instead of handling it directly inside of `AnsiTransactionManager::commit_transaction`?

Reply:
No you're right that was a mistake